### PR TITLE
riva nmt streaming s2t client update

### PR DESCRIPTION
**Pull request by ydharavath**
Previously, when processing a folder of audio files using riva nmt streaming s2t client, the output in output.txt is overwritten with each new file processed. This results in only the last audio file's output being retained in the file.

So now the client is updated so that the files are not overwritten but appended and the output will now be shown in json format in the output file

### CODE REVIEW

File: riva/clients/nmt/riva_nmt_t2t_client.cc
Comment (rmittal-github) [inline riva/clients/nmt/riva_nmt_t2t_client.cc:1]: can we remove these changes from this PR? I think they are for another issue and tracked in https://github.com/nvidia-riva/cpp-clients/pull/90, right?

Comment (ydharavath) [reply riva/clients/nmt/riva_nmt_t2t_client.cc:1]: Yeah, we can remove these changes

File: riva/clients/nmt/streaming_s2t_client.cc [hunk 1]
```diff
@@ -218,7 +218,7 @@ StreamingS2TClient::DoStreamingFromFile(
   while (true) {
     while (NumActiveStreams() < (uint32_t)num_parallel_requests && all_wav_i < all_wav_max) {
       std::unique_ptr<Stream> stream(new Stream(all_wav_repeated[all_wav_i], all_wav_i));
-      StartNewStream(std::move(stream));
+      StartNewStream(std::move(stream), all_wav_repeated[all_wav_i]->filename);
```
Comment (rmittal-github) [inline riva/clients/nmt/streaming_s2t_client.cc]: do we need to pass the filename through all these functions? is it possible to move to similar way as ASR where we are writing the file from client_call.cc which has access to the filename? In S2T as well, we get PostProcessResults which can call a ClientCall function to write the file.

**Comment by rmittal-github** [comment]
@ydharavath let's also rebase to latest top of tree.

File: riva/clients/nmt/streaming_s2t_client.cc (hunk 1 above)
Comment (ydharavath) [reply riva/clients/nmt/streaming_s2t_client.cc]: Looking into this

File: riva/clients/nmt/streaming_s2t_client.cc [hunk 2]
```diff
@@ -53,9 +53,9 @@ MicrophoneThreadMain(
 StreamingS2TClient::StreamingS2TClient(
     std::shared_ptr<grpc::Channel> channel, int32_t num_parallel_requests,
     const std::string& source_language_code, const std::string& target_language_code,
-    const std::string& dnt_phrases_file, bool profanity_filter, bool automatic_punctuation,
-    bool separate_recognition_per_channel, int32_t chunk_duration_ms, bool simulate_realtime,
-    bool verbatim_transcripts, const std::string& boosted_phrases_file, float boosted_phrases_score,
+    bool profanity_filter, bool automatic_punctuation, bool separate_recognition_per_channel,
```
Comment (rmittal-github) [inline riva/clients/nmt/streaming_s2t_client.cc:56]: still contains these unwanted changes

File: riva/clients/nmt/riva_nmt_streaming_s2t_client.cc [hunk 3]
```diff
@@ -124,7 +129,7 @@ main(int argc, char** argv)
 
   std::signal(SIGINT, signal_handler);
   gflags::ParseCommandLineFlags(&argc, &argv, true);
-
+  emptyFileBeforeRequest();
```
Comment (rmittal-github) [inline riva/clients/nmt/riva_nmt_streaming_s2t_client.cc:132]: do we need this function? at the start of fresh run of client, new file should get created that would overwrite any previous content. We should not need to explicitly empty previously existing file.

**Comment by rmittal-github** [comment]
replaced with https://github.com/nvidia-riva/cpp-clients/pull/99, closing this one

**Pull request closed without merging**